### ISSUE TITLE
post_soft_action_delay not actually used

### ISSUE THREAD

**Issue by gojkoz** [hakavlad/nohang#106]
I cannot see that post_soft_action_delay is actually being used to sleep after a soft corrective action.
Reviewed the code and only place I can find is here:
https://github.com/hakavlad/nohang/blob/cf6b213afe32cedb747785c15791b5ae2169659b/src/nohang#L2710

Might be indenting problem, maybe this was intended to be placed outside of the **except** block?

**Comment by hakavlad** [comment]
Good catch, thanks.

Seems like `post_soft_action_delay` is not work as as intended, but it is not a big problem.

I will think about how to fix this later.

```
         sleep(post_soft_action_delay) 
```
-- this line may be safety removed. This is not how it should work.

>maybe this was intended to be placed outside of the except block?

No, the problem should be solved differently.

**Comment by gojkoz** [comment]
Thanks for confirming. I am going to patch my local copy for now but likely not in a way others will want to use it. For the specific use case I'm trying I just need it to **not do anything for some time, after it's implemented one soft corrective action**. I want to block further checking for a defined period - prevent it from reacting too often on a runaway process that's re-spawning. Server use case. If you can cover that as an option in a future release, that would be great.

Good tool otherwise, thanks for making it.

**Comment by hakavlad** [comment]
Do you have any problem with current version?

>to not do anything for some time, after it's implemented one soft corrective action

Nohang does nothing after implementing soft corrective action, it waits until the victim dies. If the victim does not respond on soft corrective action, nohang sends SIGKILL (after `max_soft_exit_time`, 10s by default).

**Comment by hakavlad** [comment]
In fact, after soft corrective action, if `victim_cache_time = 10` and `max_soft_exit_time = 10` (deafault values) nohang will do nothing 10s if hard threshold is not exceeded.

**Comment by gojkoz** [comment]
You are right, I didn't know about victim_cache_time. That's what I need actually, to not touch the same victim too often even if it's to blame. That might be the easiest solution for what I need, I was trying to do it with the post_soft_action_delay at first (delay doing anything again after an action).

**Comment by hakavlad** [comment]
>not touch the same victim too often

nohang sends SIGTERM to one process only once. The next signal is SIGKILL after `soft_exit_time`, if the process does not respond on SIGKILL.

`victim_cache_time` means the time during which nohang should not look for a new victim if the old victim is alive.
